.github/workflows/job_PreCommit_Portable_Python.yml

```diff
@@ -0,0 +1,67 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Portable Python
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ['model/**', 'runners/core-construction-java/**', 'runners/core-java/**', 'runners/extensions-java/**', 'runners/flink/**', 'runners/java-fn-execution/**', 'runners/reference/**', 'sdks/python/**', 'release/**']
+  pull_request:
+    branches: ['master', 'release-*']
+    paths: ['model/**', 'runners/core-construction-java/**', 'runners/core-java/**', 'runners/extensions-java/**', 'runners/flink/**', 'runners/java-fn-execution/**', 'runners/reference/**', 'sdks/python/**', 'release/**']
+
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '* */6 * * *'
+
+jobs:
+  PortablePythonPreCommit:
+    name: Portable Python
+    runs-on: ['self-hosted', ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix:
+        python_version: ['3.8', '3.11']
+    if: github.event_name == 'push'|| github.event_name == 'pull_request'|| github.event_name == 'schedule'|| github.event.comment.body == 'Run Portable_Python PreCommit'
+
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+      - name: Install Java
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
```

Comment:
Can we use the `Setup self-hosted` step instead of `Install Java` and `Install Go` steps?
```
- name: Setup self-hosted
   uses: ./.github/actions/setup-self-hosted-action
   with:
     requires-py-39: false
```

For Python installation, I think it's fine to leave it as a matrix here or we can extend the `./.github/actions/setup-self-hosted-action/action.yml` file by adding this matrix there